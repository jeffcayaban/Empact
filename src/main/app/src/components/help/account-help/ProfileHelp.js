import React from "react";
import {Col, Grid, Row} from "react-bootstrap";

import ProfileLink from '../../../other/images/profile-link.png';
import ProfilePage from '../../../other/images/profile-page.png';
import BackButton from "../../common/BackButton";

/**
 * Displays help information for a profile page.
 */

class ProfileHelp extends React.PureComponent {

    render() {
        return (
            <div className={'secondaryPageContainerStyle'}>
                <div className={"centerStyle"}>
                    <h2>Your Profile</h2>
                    <p>Understand the different features of your profile page.</p>
                </div>
                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"}>
                        <Row>
                            <Col md={12} lg={12}>
                                <div>
                                    <h4 style={{ margin: 0 }}>Your profile page</h4>
                                    <hr className={"helpHr"} />

                                    <p>
                                        Go to your profile page by clicking on the "Profile" link. This link is found
                                        on the top navigation bar. You can additionally view another user's profile page
                                        by clicking on their name on an argument or petition.
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingTop: '20px', paddingBottom: '20px' }}>
                                        <img style={{ maxWidth: '30%', minWidth: '350px', align: 'center' }}
                                             src={ProfileLink}
                                             alt="Profile Page Link"
                                        />
                                    </div>

                                    <p>
                                        Below shows an image of a typical profile page. It features an area for your
                                        petitions and another area for your arguments. 
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingTop: '20px', paddingBottom: '20px' }}>
                                        <img style={{ maxWidth: '80%', minWidth: '350px', align: 'center' }}
                                             src={ProfilePage}
                                             alt="Profile Page"
                                        />
                                    </div>
                                </div>
                                <BackButton link={"/help/account"}/>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default ProfileHelp;