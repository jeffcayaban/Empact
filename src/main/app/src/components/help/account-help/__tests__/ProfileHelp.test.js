import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import ProfileHelp from "../ProfileHelp";

describe('ProfileHelp', () => {

    it('should render successfully', () => {
        const component = shallow(<ProfileHelp />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
