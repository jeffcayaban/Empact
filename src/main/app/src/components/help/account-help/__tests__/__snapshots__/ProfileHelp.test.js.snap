// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHelp should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2>
      Your Profile
    </h2>
    <p>
      Understand the different features of your profile page.
    </p>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
        >
          <div>
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              Your profile page
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              Go to your profile page by clicking on the "Profile" link. This link is found on the top navigation bar. You can additionally view another user's profile page by clicking on their name on an argument or petition.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <img
                alt="Profile Page Link"
                src="profile-link.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "30%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
            <p>
              Below shows an image of a typical profile page. It features an area for your petitions and another area for your arguments.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <img
                alt="Profile Page"
                src="profile-page.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "80%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
          </div>
          <BackButton
            link="/help/account"
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
